src/main/java/org/ibp/api/brapi/v2/list/GermplasmListResourceBrapi.java

```diff
@@ -0,0 +1,102 @@
+import org.ibp.api.domain.common.PagedResult;
+import org.ibp.api.rest.common.PaginatedSearch;
+import org.ibp.api.rest.common.SearchSpec;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Api(value = "BrAPI v2 Germplasm List Services")
+@Controller(value = "GermplasmListResourceBrapi")
+public class GermplasmListResourceBrapi {
+
+	private final static String ALLOWED_LIST_TYPE = "germplasm";
+
+	@Autowired
+	private GermplasmListServiceBrapi germplasmListServiceBrapi;
+
+	@ApiOperation(value = "Get filtered set of generic lists", notes = "Get filtered set of generic lists")
+	@RequestMapping(value = "/{crop}/brapi/v2/lists", method = RequestMethod.GET)
+	@JsonView(BrapiView.BrapiV2.class)
+	@ResponseBody
+	public ResponseEntity<EntityListResponse<GermplasmListDTO>> getSamples(@PathVariable final String crop,
+		@ApiParam(value = "The type of objects contained by this generic list")
+		@RequestParam(value = "listType", required = false) final String listType,
+		@ApiParam(value = "The human readable name of this generic list")
+		@RequestParam(value = "listName", required = false) final String listName,
+		@ApiParam(value = "The unique ID of this generic list")
+		@RequestParam(value = "listDbId", required = false) final String listDbId,
+		@ApiParam(value = "The source tag of this generic list")
+		@RequestParam(value = "listSource", required = false) final String listSource,
+		@ApiParam(value = "An external reference ID. Could be a simple string or a URI. (use with externalReferenceSource parameter)")
+		@RequestParam(value = "externalReferenceID", required = false) final String externalReferenceID,
+		@ApiParam(value = "An identifier for the source system or database of an external reference (use with externalReferenceID parameter)")
+		@RequestParam(value = "externalReferenceSource", required = false) final String externalReferenceSource,
+		@ApiParam(value = BrapiPagedResult.CURRENT_PAGE_DESCRIPTION, required = false)
+		@RequestParam(value = "page", required = false) final Integer currentPage,
+		@ApiParam(value = BrapiPagedResult.PAGE_SIZE_DESCRIPTION, required = false)
+		@RequestParam(value = "pageSize", required = false) final Integer pageSize) {
+
+		if(StringUtils.isNotEmpty(listType) && !ALLOWED_LIST_TYPE.equalsIgnoreCase(listType)) {
+			return new ResponseEntity<>(new EntityListResponse<>(new Result<>(new ArrayList<>())), HttpStatus.OK);
```

Comment:
Why are we returning a http 200 when list type is not supported? Should be http 400 (Bad request) but with proper error message.